From: Ian Ferguson <iferguso@genome.wustl.edu>
Date: Fri, 27 Feb 2015 17:27:36 -0600
Subject: [PATCH] Tweak install destinations in Makefile

Use version as suffix for binaries, move miscellaneous programs to subdirectory
of /usr/share
---
 Makefile |   11 ++++++-----
 1 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index 21b8c6c..339a107 100644
--- a/Makefile
+++ b/Makefile
@@ -39,9 +39,10 @@ AOBJS=      bam_index.o bam_plcmd.o sam_view.o \
 INCLUDES=   -I. -I$(HTSDIR)
 LIBCURSES=  -lcurses # -lXCurses
 
-prefix      = /usr/local
+prefix      = /usr
 exec_prefix = $(prefix)
 bindir      = $(exec_prefix)/bin
+sharedir    = $(prefix)/share/samtools
 mandir      = $(prefix)/share/man
 man1dir     = $(mandir)/man1
 
@@ -59,7 +60,6 @@ BUILT_MISC_PROGRAMS = \
 	misc/md5fa misc/md5sum-lite misc/wgsim
 
 MISC_PROGRAMS = \
-	$(BUILT_MISC_PROGRAMS) \
 	misc/blast2sam.pl misc/bowtie2sam.pl misc/export2sam.pl \
 	misc/interpolate_sam.pl misc/novo2sam.pl \
 	misc/plot-bamstats misc/psl2sam.pl \
@@ -256,9 +256,10 @@ misc/md5sum-lite.o: misc/md5.c misc/md5.h
 
 
 install: $(PROGRAMS) $(BUILT_MISC_PROGRAMS)
-	$(INSTALL_DIR) $(DESTDIR)$(bindir) $(DESTDIR)$(man1dir)
-	$(INSTALL_PROGRAM) $(PROGRAMS) $(MISC_PROGRAMS) $(DESTDIR)$(bindir)
-	$(INSTALL_DATA) samtools.1 $(DESTDIR)$(man1dir)
+	$(INSTALL_DIR) $(DESTDIR)$(bindir) $(DESTDIR)$(man1dir) $(DESTDIR)$(sharedir)$(PACKAGE_VERSION)
+	$(foreach program,$(PROGRAMS) $(BUILD_MISC_PROGRAMS),$(INSTALL_PROGRAM) -T $(program) $(DESTDIR)$(bindir)/$(program)$(PACKAGE_VERSION);)
+	$(INSTALL_PROGRAM) $(MISC_PROGRAMS) $(DESTDIR)$(sharedir)$(PACKAGE_VERSION)
+	$(INSTALL_DATA) -T samtools.1 $(DESTDIR)$(man1dir)/samtools$(PACKAGE_VERSION).1
 
 
 testclean:
-- 
